File: maxNum/maxNum.py
import itertools

def solution(numbers):
  arr = itertools.permutations(numbers,len(numbers))
  arr = list(arr)
  string = ''
  candidate = []
  length = 0

  for i in numbers:
    length += len(str(i))
  for i in range(len(arr)):
    for j in range(len(arr[i])):
      string += str(arr[i][j])
      if len(string) == length:
        candidate.append(string)
        string = '';
  candidate.sort(reverse=True)
  answer = str(int(candidate[0]))
  return answer

print(solution([3, 30, 34, 5, 9]))

File: maxNum/maxNum.js
function getCombinations(arr, selected) {
  const result = [];
  if (selected === 1) return arr.map((el) => [el]);

  arr.forEach((fixed, idx, origin) => {
    const rest = origin.slice(idx + 1);
    const combination = getCombinations(rest, selected - 1);
    const addition = combination.map((el) => [fixed, ...el]);
    result.push(...addition);
  });
  return result;
}

function solution(numbers) {
  var answer = '';
  let max = Number.MAX_SAFE_INTEGER;
  console.log(getCombinations(numbers, numbers.length));
}

console.log(solution([6, 10, 2]));
// answer == 6210
